### PR TITLE
Fix: remove params from url after claiming

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description

The verified param's  (set when returning after FV) triggers a claim request.
The param is never removed so it gets into a repeated claim-request loop whenever you switch pages etc.

Now after a claim, we clear the URL to its default location

About # (link your issue here)
https://github.com/GoodDollar/GoodProtocolUI/issues/357

### CODE REVIEW

File: packages/good-design/src/core/buttons/ClaimButton.tsx [hunk 1]
```diff
@@ -214,6 +213,9 @@ const ClaimButton = ({
         showActionModal();
         await handleClaim();
         setClaimLoading(true);
+        // Here we assume claiming was completed succesfully,
+        // so it pushes the default route without params to prevent repeated claim requests
+        window.history.pushState(null, "", window.location.origin + window.location.hash);
```
Comment (johnsmith-gooddollar) [inline packages/good-design/src/core/buttons/ClaimButton.tsx:218]: it's a **dirty** workaround

first-of-all we're making our code native base e.g. cross-platform. this fixes web only 
second thing - react apps have own navigation state management so any library shouldn't manipulate it **directly**

**this should be reverted**

do such redirect at protocol UI side (via nav library used there)

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: packages/good-design/src/core/buttons/ClaimButton.tsx (hunk 1 above)
Comment (L03TJ3) [reply packages/good-design/src/core/buttons/ClaimButton.tsx:218]: Its not doing a redirect, above just clears history without refresh
But if we need to then yes, we can do it

Comment (johnsmith-gooddollar) [reply packages/good-design/src/core/buttons/ClaimButton.tsx:218]: @L03TJ3 this shouldn't be here, and shouldn't be done through raw API.

Comment (johnsmith-gooddollar) [reply packages/good-design/src/core/buttons/ClaimButton.tsx:218]: pls fix ASAP

Comment (L03TJ3) [reply packages/good-design/src/core/buttons/ClaimButton.tsx]: fixed, don't refresh but just don't execute when claimed

**Review by johnsmith-gooddollar — APPROVED** [review]

**Pull request merged**